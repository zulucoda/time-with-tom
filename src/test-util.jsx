import React from 'react';
import { oneOfType, shape, element, func } from 'prop-types';
import { Form, Formik } from 'formik';
import { BrowserRouter as Router } from 'react-router-dom';

/**
 * Formik Wrapper Test Util
 * @param {React.FunctionComponent} Component
 * @param {Object} props
 * @param {Object} initialValues
 * @param {Function} onSubmit
 * @returns {Formik}
 */
export const FormikWrapperTestUtil = ({
  Component,
  props = {},
  initialValues = {},
  onSubmit = jest.fn(),
}) => (
  <Formik initialValues={initialValues} onSubmit={onSubmit}>
    <Form>
      <Component {...props} />
    </Form>
  </Formik>
);

FormikWrapperTestUtil.propTypes = {
  Component: oneOfType([element.isRequired, func]),
  props: shape({}),
  initialValues: shape({}),
  onSubmit: func,
};

/**
 * React Router Wrapper Test Util
 * @param {React.FunctionComponent} Component
 * @param {object} props
 * @returns {React.FunctionComponent}
 */
export const ReactRouterWrapperTestUtil = ({ Component, props = {} }) => (
  <Router>
    <Component {...props} />
  </Router>
);

ReactRouterWrapperTestUtil.propTypes = {
  Component: oneOfType([element.isRequired, func]),
  props: shape({}),
};
