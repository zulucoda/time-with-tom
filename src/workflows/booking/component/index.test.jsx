import { render } from '@testing-library/react';
import { FormikWrapperTestUtil } from '../../../test-util';
import Booking from './index';
import React from 'react';

describe('Workflows - Booking - Component - Unit Test', () => {
  test('does it render', () => {
    const { container } = render(<FormikWrapperTestUtil Component={Booking} />);
    expect(container).toMatchSnapshot();
  });
});
