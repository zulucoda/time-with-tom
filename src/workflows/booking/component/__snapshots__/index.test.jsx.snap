// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Workflows - Booking - Component - Unit Test does it render 1`] = `
<div>
  <form
    action="#"
  >
    <div>
      <div
        class="sc-AxirZ cbHwXd"
      >
        <div
          class="MuiFormControl-root MuiTextField-root"
          format="yyyy/MM/dd"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
            data-shrink="true"
          >
            date
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input"
              name="date"
              placeholder="2020/04/21"
              readonly=""
              type="text"
              value="2020/04/21"
            />
          </div>
          
        </div>
      </div>
      <div
        class="sc-AxiKw grOufF"
      >
        <div
          class="MuiFormControl-root MuiTextField-root"
          format="HH:mm"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
            data-shrink="true"
          >
            startTime
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input"
              name="startTime"
              placeholder="18:00"
              readonly=""
              type="text"
              value="02:00"
            />
          </div>
          
        </div>
      </div>
      <div
        class="sc-AxiKw grOufF"
      >
        <div
          class="MuiFormControl-root MuiTextField-root"
          format="HH:mm"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
            data-shrink="true"
          >
            endTime
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input"
              name="endTime"
              placeholder="18:00"
              readonly=""
              type="text"
              value="02:00"
            />
          </div>
          
        </div>
      </div>
    </div>
  </form>
</div>
`;
